onnxruntime/core/platform/windows/debug_alloc.cc

```diff
@@ -232,12 +232,8 @@ Memory_LeakCheck::~Memory_LeakCheck() {
     _snprintf_s(buffer, _TRUNCATE, "%d bytes of memory leaked in %d allocations", leaked_bytes, leak_count);
     string.append(buffer);
 
-    // If we're being actively debugged, show a message box to get the dev's attention
-    if (IsDebuggerPresent())
-      MessageBoxA(nullptr, string.c_str(), "Warning", MB_OK | MB_ICONWARNING);
-    else {
-      // If we're on the command line (like on a build machine), output to the console and exit(-1)
-      std::cout << "\n----- MEMORY LEAKS: " << string.c_str() << "\n";
+    std::cout << "\n----- MEMORY LEAKS: " << string.c_str() << "\n";
+    if (!IsDebuggerPresent()) {
```

Comment:
@RyanUnderhill  FYI.

Reply:
What's the rationale for removing the message box? I found it pretty useful.

Reply:
It links to desktop only libs. I removed all desktop only libs from the default link libraries and this was the only piece of code that required a desktop lib. This was an effort not to depend on desktop and to use onecoreuap libs.

We could conditionally link to desktop libs only on debug builds, but I thought it was easier to just remove this one message box.

Reply:
BTW, I'm trying to add a macro to control this memory leak checker:
https://github.com/microsoft/onnxruntime/commit/f85420fedbbb345e502e2b455129fa86cf1ed3f9#diff-83ba5c7244b867af5d86acd34e43f7ca

```c++
#ifdef ONNXRUNTIME_ENABLE_MEMLEAK_CHECK
```

Reply:
@tiagoshibata Can you please take a look at this feedback? It seems that the debugger present check may not be needed here?

Reply:
I didn't add the `IsDebuggerPresent`, it already existed. What was the rationale behind exiting early if a debugger is not present @RyanUnderhill ?